File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.js
const input = require('fs').readFileSync(0, 'utf-8').toString().trim().split('\n');
const n = +input[0];
const logs = input.slice(1).map(x => x.split(' '));
const set = new Set();

for (let log of logs) {
    const [name, action] = log;
    
    if (action === 'enter') {
        set.add(name);
    } else {
        set.delete(name);
    }
}

const answer = Array.from(set).sort().reverse();

console.log(answer.join('\n'));

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys

input = sys.stdin.readlines()
S = set()

for line in input[1:]:
    name, event = line.split()
    
    if (event == 'enter'):
        S.add(name)
    else:
        S.discard(name)

print('\n'.join(sorted(S, reverse=True)))